thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/detection/AnomalyDetectionInputContextBuilder.java

```diff
@@ -0,0 +1,640 @@
+    this.anomalyFunction = anomalyFunctionFactory.fromSpec(anomalyFunctionSpec);
+    this.anomalyDetectionInputContext = anomalyDetectionInputContext;
+    this.dataset = this.anomalyFunctionSpec.getCollection();
+    DatasetConfigDTO datasetConfig = DAO_REGISTRY.getDatasetConfigDAO().findByDataset(dataset);
+    if (datasetConfig == null) {
+      LOG.error("Dataset [" + dataset + "] is not found");
+      throw new NullArgumentException(
+          "Dataset [" + dataset + "] is not found with function : " + anomalyFunctionSpec
+              .toString());
+    }
+    this.collectionDimensions = datasetConfig.getDimensions();
+    return this;
+  }
+
+  public AnomalyDetectionInputContext build() {
+    return this.anomalyDetectionInputContext;
+  }
+
+  /**
+   * Fetch TimeSeriese data from Pinot in the startEndTimeRanges
+   * @param startEndTimeRanges
+   * @return a AnomalyDetectionInputContextBuilder
+   * @throws JobExecutionException
+   * @throws ExecutionException
+   */
+  public AnomalyDetectionInputContextBuilder fetchTimeSeriesData(List<Pair<Long, Long>> startEndTimeRanges, boolean endTimeInclusive)
+      throws JobExecutionException, ExecutionException {
+    Map<DimensionKey, MetricTimeSeries> dimensionKeyMetricTimeSeriesMap =
+        getTimeSeriesForAnomalyDetection(anomalyFunctionSpec, startEndTimeRanges, endTimeInclusive);
+
+    Map<DimensionMap, MetricTimeSeries> dimensionMapMetricTimeSeriesMap = new HashMap<>();
+    for (Map.Entry<DimensionKey, MetricTimeSeries> entry : dimensionKeyMetricTimeSeriesMap.entrySet()) {
+      DimensionKey dimensionKey = entry.getKey();
+
+      // If the current time series belongs to OTHER dimension, which consists of time series whose
+      // sum of all its values belows 1% of sum of all time series values, then its anomaly is
+      // meaningless and hence we don't want to detection anomalies on it.
+      String[] dimensionValues = dimensionKey.getDimensionValues();
+      boolean isOTHERDimension = false;
+      for (String dimensionValue : dimensionValues) {
+        if (dimensionValue.equalsIgnoreCase(ResponseParserUtils.OTHER) || dimensionValue.equalsIgnoreCase(ResponseParserUtils.UNKNOWN)) {
+          isOTHERDimension = true;
+          break;
+        }
+      }
+      if (isOTHERDimension) {
+        continue;
+      }
+
+      DimensionMap dimensionMap = DimensionMap.fromDimensionKey(dimensionKey, collectionDimensions);
+      dimensionMapMetricTimeSeriesMap.put(dimensionMap, entry.getValue());
+
+      if (entry.getValue().getTimeWindowSet().size() < 1) {
+        LOG.warn("Insufficient data for {} to run anomaly detection function", dimensionMap);
+        continue;
+      }
+    }
+    this.anomalyDetectionInputContext.setDimensionKeyMetricTimeSeriesMap(dimensionMapMetricTimeSeriesMap);
+
+    return this;
+  }
+  /**
+   * Fetch TimeSeriese data from Pinot with given monitoring window start and end time
+   * @param windowStart
+   * @param windowEnd
```

Comment:
Add comment to indicate that this method is end time exclusive unless the entTimeInclusive flag is set. Moreover, the start time is inclusive.